src/Agent.ps1

```diff
@@ -1,58 +1,66 @@
-              return $agentPid
-          }
-          else {
-              setenv 'SSH_AGENT_PID' $null
-              setenv 'SSH_AUTH_SOCK' $null
-          }
-      }
-  }
-
-  return 0
+    if ($env:GIT_SSH -imatch 'plink') {
+        $pageantPid = Get-Process | Where-Object { $_.Name -eq 'pageant' } | Select-Object -ExpandProperty Id -First 1
+        if ($null -ne $pageantPid) { return $pageantPid }
+    }
+    elseif ($native = Get-NativeSshAgent) {
+        return $native
+    }
+    else {
+        $agentPid = $Env:SSH_AGENT_PID
+        if ($agentPid) {
+            $sshAgentProcess = Get-Process | Where-Object { ($_.Id -eq $agentPid) -and ($_.Name -eq 'ssh-agent') }
+            if ($null -ne $sshAgentProcess) {
+                return $agentPid
+            }
+            else {
+                setenv 'SSH_AGENT_PID' $null
+                setenv 'SSH_AUTH_SOCK' $null
+            }
+        }
+    }
+
+    return 0
 }
 
 # Attempt to guess $program's location. For ssh-agent/ssh-add.
 function Find-Ssh($program = 'ssh-agent') {
-  Write-Verbose "$program not in path. Trying to guess location."
-  $gitItem = Get-Command git -CommandType Application -Erroraction SilentlyContinue | Get-Item
-  if ($null -eq $gitItem) {
-      Write-Warning 'git not in path'
-      return
-  }
-
-  $sshLocation = join-path $gitItem.directory.parent.fullname bin/$program
-  if (get-command $sshLocation -Erroraction SilentlyContinue) {
-      return $sshLocation
-  }
-
-  $sshLocation = join-path $gitItem.directory.parent.fullname usr/bin/$program
-  if (get-command $sshLocation -Erroraction SilentlyContinue) {
-      return $sshLocation
-  }
+    Write-Verbose "$program not in path. Trying to guess location."
+    $gitItem = Get-Command git -CommandType Application -Erroraction SilentlyContinue | Get-Item
+    if ($null -eq $gitItem) {
+        Write-Warning 'git not in path'
+        return
+    }
+
+    $sshLocation = join-path $gitItem.directory.parent.fullname bin/$program
+    if (get-command $sshLocation -Erroraction SilentlyContinue) {
+        return $sshLocation
+    }
+
+    $sshLocation = join-path $gitItem.directory.parent.fullname usr/bin/$program
+    if (get-command $sshLocation -Erroraction SilentlyContinue) {
+        return $sshLocation
+    }
 }
 
 # Loosely based on bash script from http://help.github.com/ssh-key-passphrases/
 function Start-SshAgent {
     param(
-       [Parameter(Position = 0)]
-       [ValidateSet("Automatic", "Boot", "Disabled", "Manual", "System")]
-       [string]
-       $StartupType = "Manual",
-
-       [Parameter()]
-       [switch]
-       $Quiet
+        [Parameter(Position = 0)]
+        [ValidateSet("Automatic", "Boot", "Disabled", "Manual", "System")]
+        [string]
+        $StartupType = "Manual",
+
+        [Parameter()]
+        [switch]
+        $Quiet,
+
+        [Parameter()]
+        [ValidateSet("Machine", "Process", "User")]
+        [string]
+        $Scope = "Process"
```

Comment:
Is this parameter being used?

Reply:
It is now! :) I forgot to commit the change